src/hotspot/share/gc/g1/g1CollectedHeap.hpp

```diff
@@ -145,7 +145,7 @@ class G1CollectedHeap : public CollectedHeap {
 private:
   G1ServiceThread* _service_thread;
   G1ServiceTask* _periodic_gc_task;
-  G1SegmentedArrayFreeMemoryTask* _free_segmented_array_memory_task;
+  G1MonotonicArenaFreeMemoryTask* _free_monotonic_arena_memory_task;
```

Comment:
I do not think it is required to have variable names that repeat the whole type name, particularly if there is no ambiguity. I.e. `_free_arena_memory_task` seems sufficient instead of `_free_monotonic_arena....` but I do not mind either way.